distribution/push.go

```diff
@@ -127,6 +133,19 @@ func Push(ctx context.Context, ref reference.Named, imagePushConfig *ImagePushCo
 			continue
 		}
 
+		parsedURL, urlErr := url.Parse(endpoint.URL)
```

Comment:
@calavera: Do you prefer this approach? https://gist.github.com/aaronlehmann/dcec73055470ba81cd4f

I'm not sure I like the set of function arguments and return values:

    func checkRegistryIncompatibility(endpoint registry.APIEndpoint, confirmedTLSRegistries map[string]struct{}, confirmedV2 bool) (*url.URL, error) {

...but I don't really have strong feelings about this.